src/rust/ide/src/controller/visualization.rs

```diff
@@ -0,0 +1,251 @@
+//! Visualization controller.
+//!
+//! Ths Visualization Controller is Responsible identifying all the available visualizations
+//! natively embedded in IDE and available within the project's `visualization` folder.
+
+use crate::prelude::*;
+
+use enso_protocol::language_server;
+use graph_editor::component::visualization::class;
+use graph_editor::component::visualization::JsSourceClass;
+use std::rc::Rc;
+
+
+
+// =============
+// === Error ===
+// =============
+
+/// Enumeration of errors used in `Visualization Controller`.
+#[derive(Debug,Fail)]
+#[allow(missing_docs)]
+pub enum VisualizationError {
+    #[fail(display = "Visualization \"{}\" not found.", identifier)]
+    NotFound {
+        identifier : VisualizationPath
+    },
+    #[fail(display = "JavaScript visualization \"{}\" failed to be instantiated.", identifier)]
+    InstantiationError {
+        identifier : VisualizationPath
+    }
+}
+
+
+
+// =========================
+// === VisualizationPath ===
+// =========================
+
+/// This enum is used to provide a path to visualization either in the project folder or natively
+/// embedded in IDE.
+#[derive(Clone,Debug,Display,Eq,PartialEq)]
+#[allow(missing_docs)]
+pub enum VisualizationPath {
+    Embedded(String),
+    File(language_server::Path)
+}
+
+
+
+// ==============================
+// === EmbeddedVisualizations ===
+// ==============================
+
+#[allow(missing_docs)]
+pub type EmbeddedVisualizationName = String;
+
+/// Embedded visualizations mapped from name to source code.
+#[derive(Shrinkwrap,Debug,Clone,Default)]
+#[shrinkwrap(mutable)]
+pub struct EmbeddedVisualizations {
+    #[allow(missing_docs)]
+    pub map:HashMap<EmbeddedVisualizationName,Rc<class::Handle>>
+}
+
+
+
+
+// ==============
+// === Handle ===
+// ==============
+
+const VISUALIZATION_FOLDER:&str = "visualization";
```

Comment:
Oh no, please no. You are defining a folder constant in a module `controller::visualization`. This is very, very bad. Such constants should be in some top-level file, easily accessible and visible.